rn/Teacher/src/modules/inbox/components/ConversationMessageRow.js

```diff
@@ -71,7 +72,9 @@ export default class ConversationMessageRow extends React.Component<Props, State
       }
       return id === author_id
     })
-    return Object.keys(target?.common_courses ?? {})[0]
+    return Object.keys(target?.common_courses ?? {}).filter(id => (
+      contextType !== 'course' || id === contextID
+    ))[0]
```

Comment:
The filter with the hard indexing scares me. Will the filtered list always have at least one element?

Reply:
no, but it will just return undefined in that case. It won't crash like it would in Swift.